client/consensus/slots/src/lib.rs

```diff
@@ -121,11 +121,10 @@ pub trait SimpleSlotWorker<B: BlockT> {
 			StorageChanges<<Self::BlockImport as BlockImport<B>>::Transaction, B>,
 			Self::Claim,
 			Self::EpochData,
-		) -> sp_consensus::BlockImportParams<
-			B,
-			<Self::BlockImport as BlockImport<B>>::Transaction
-		>
-		+ Send
+		) -> Result<
```

Comment:
Why is this returning a result now?

Reply:
Multiple scenarios could happen here:
1. Errors could be returned by calling `sign_with` which itself returns a result. Thinking of this in the context of having the signing done remotely, then errors could popup from doing that.
2. Converting the returned signature(`vec<u8>`) into our expected signature type could also fail.